### DISCUSSION TITLE
The case of the checks (INT versus Int versus int)

### DISCUSSION THREAD

**Discussion by Ovid** [Perl-Oshun/oshun#8] [Ideas]
The reason we went with SHOUTY types (`INT`, `HASH`, and so on) is because it was a subtle disaffordance to using native types. We'd want to encourage people to create _meaningful_ types. Instead of this:

```perl
sub f_to_c ($f :of(NUM)) :returns(NUM) {...}
```

We'd encourage this:

```perl
sub f_to_c ($f :of(Fahrenheit) :returns(Celsius) {...}
```

And in the type definitions:

```perl
check Celsius    :isa(NUM) ($value) { $value >= -273.14 }
check Fahrenheit :isa(NUM) ($value) { $value >= −459.67 }
```

Or more succinctly (and more performant):

```perl
check Celsius    :isa(NUM[-273.14..inf]);
check Fahrenheit :isa(NUM[−459.67..inf]);
```

However, "SHOUTY" checks has been one of the primary sources of complaints about the syntax and honestly, they annoy the heck out of me, too.

In reviewing code bases using `Type::Tiny`, I often find people resorting just to the bundled types because they have a "one-off" signature that doesn't justify the burden of declaring a new check. And honestly, SHOUTY doesn't feel very Perlish.

My current thought it to do something similar to Java. Native types would be lower-case and user-defined would have to start with an upper-case ASCII letter.

```perl
my @temps :of(hash[hash[Celsius]]) = get_temperatures();
```

That certainly violates the original intent, but the aesthetics are better (IMHO) and the clear distinction between user and native types would remain.

**Comment by Ovid** [comment]
Do collaborators get automatic notification on something like this? I hope so. It seems like this could be one of the easier issues to resolve. Would love to hear people's thoughts.

**Reply by rabbiveesh** [reply]
You need to specifically watch the repo or the issue; you don't get notified by default unless you @ mention them (is this a GitHub org? You could probably make a team for group mentions)

**Reply by rwp0** [reply]
Please transform this repository into an organization in the future if possible.

You can create an org and then transfer this repo into it.
